services/construction_service.go

```diff
@@ -248,7 +267,14 @@ func (s *ConstructionAPIService) ConstructionMetadata(
 		return nil, wrapErr(ErrGeth, err)
 	}
 
-	gasLimit := optimism.TransferGasLimit
+	var gasLimit uint64
+	if input.GasLimit == nil {
+		// by default, initialize gasLimit to the TransferGasLimit
+		gasLimit = optimism.TransferGasLimit
+	} else {
+		gasLimit = input.GasLimit.Uint64()
```

Comment:
For big.Uint64(), it says the result is undefined if "x" cannot be represented in a uint64. Should we check for overflow just in case and if so set it to math.MaxUint64 ?

Reply:
Is there a standard library/way in golang to check for overflow on conversion?